mutt/notify.c

```diff
@@ -37,11 +37,11 @@
 /// Lookup table for NotifyType
 /// Must be the same size and order as #NotifyType
 static const char *NotifyTypeNames[] = {
-  "NT_ALL",     "NT_ACCOUNT",  "NT_ALIAS",   "NT_ALTERN", "NT_ATTACH",
-  "NT_BINDING", "NT_COLOR",    "NT_COMMAND", "NT_CONFIG", "NT_CONTEXT",
-  "NT_EMAIL",   "NT_ENVELOPE", "NT_GLOBAL",  "NT_HEADER", "NT_INDEX",
-  "NT_MAILBOX", "NT_MENU",     "NT_PAGER",   "NT_SCORE",  "NT_SUBJRX",
-  "NT_WINDOW",
+  "NT_ALL",      "NT_ACCOUNT", "NT_ALIAS",   "NT_ALTERN", "NT_ATTACH",
```

Comment:
this drudgery can be avoided by auto-generating the labels. an example can be found in https://sourceforge.net/p/isync/isync/ci/17db5de0cadf15dd52ebead3e45a0507e8c774c9/ (which requires https://sourceforge.net/p/isync/isync/ci/6a78e2c5f61fdc1ec18a87f05f800d22ca59dfa6/ (which is aimed at bitwise enums, but extending it to support regular sequences (SEQ_ENUM?) would be no big deal)).

Reply:
Since the split from Mutt, I've worked hard to eliminate lots of custom steps in the build.
We could probably copy the macro magic from opcodes.h:

https://github.com/neomutt/neomutt/blob/2f2741f62db1a7662d0b1684bd2036aab700323f/opcodes.h#L36-L37

It generates an enum and a matching string array.

Reply:
> I've worked hard to eliminate lots of custom steps in the build.

for which reasons in particular?

> We could probably copy the macro magic from opcodes.h

hmm, yeah, the "macro visitor" pattern. mildly ugly, but quite powerful.

i've meanwhile come up with a solution to automatically enumerate bits, so i could actually adapt this. thanks for the inspiration!

Reply:
> for which reasons in particular?

It was fiendishly complicated and quite fragile.
The docs build is still the worst part.
Here are two (fairly current) diagrams of our build process: https://neomutt.org/dev/build/make

And this is the original build process:

<img src="https://flatcap.org/mutt/make-orig.svg">

Reply:
i'm a build system guy, so this doesn't scare me. :joy: 

my primary concern about code generation is that tools (e.g., IDE code models) don't work right until the project was built, and even then the results might be suboptimal. but deep macro magic doesn't exactly help with debugging and IDE "follow symbol" functionality, either, so it's really a bit of a "choose your death" situation.